app/models/poll.rb

```diff
@@ -29,6 +29,7 @@ class Poll < ActiveRecord::Base
   after_touch       :ensure_status_is_correct
   after_create      :create_rating
   after_find        :close_if_expire
+  after_update      :draft!, if: :ready?
```

Comment:
Why `after_` callback? Is ti so important to change the state only after everything else is put into the DB? Otherwise you'd better use something from an earlier stage of a lifecycle — and then you'd be able to get rid of that manual reloading down the flow...